firmware/quadruna/FSM/src/tasks.c

```diff
@@ -0,0 +1,130 @@
+#include "tasks.h"
+#include "main.h"
+#include "cmsis_os.h"
+
+#include "app_globals.h"
+#include "app_heartbeatMonitor.h"
+#include "states/app_driveState.h"
+
+#include "app_canTx.h"
+#include "app_canRx.h"
+#include "app_canAlerts.h"
+#include "app_commitInfo.h"
+
+#include "io_jsoncan.h"
+
+#include "hw_bootup.h"
+#include "hw_hardFaultHandler.h"
+#include "hw_watchdog.h"
+#include "hw_watchdogConfig.h"
+
+extern ADC_HandleTypeDef  *hadc1;
+extern CAN_HandleTypeDef  *hcan1;
+extern IWDG_HandleTypeDef *hiwdg;
+extern TIM_HandleTypeDef  *htim12;
+struct StateMachine       *state_machine;
+struct HeartbeatMonitor   *heartbeat_monitor;
+
+void canRxQueueOverflowCallBack(uint32_t overflow_count)
+{
+    app_canTx_FSM_RxOverflowCount_set(overflow_count);
+    app_canAlerts_FSM_Warning_TxOverflow_set(true);
+}
+
+void canTxQueueOverflowCallBack(uint32_t overflow_count)
+{
+    app_canTx_FSM_TxOverflowCount_set(overflow_count);
+    app_canAlerts_FSM_Warning_TxOverflow_set(true);
+}
+
+const CanConfig can_config = {
+    .rx_msg_filter        = NULL,
+    .tx_overflow_callback = canTxQueueOverflowCallBack,
+    .rx_overflow_callback = canRxQueueOverflowCallBack,
+};
+
+void tasks_preInit(void) {}
+
+void tasks_init(void)
+{
+    __HAL_DBGMCU_FREEZE_IWDG();
+
+    // Configure and initialize SEGGER SystemView.
+    SEGGER_SYSVIEW_Conf();
+
+    hw_hardFaultHandler_init();
+    hw_can_init(hcan1);
+    hw_watchdog_init(hw_watchdogConfig_refresh, hw_watchdogConfig_timeoutCallback);
+
+    __HAL_DBGMCU_FREEZE_IWDG();
+
+    io_canTx_init(io_jsoncan_pushTxMsgToQueue);
+    io_canTx_enableMode(CAN_MODE_DEFAULT, true);
+    io_can_init(&can_config);
+
+    app_canTx_init();
+    app_canRx_init();
+
+    app_canTx_FSM_Hash_set(GIT_COMMIT_HASH);
+    app_canTx_FSM_Clean_set(GIT_COMMIT_CLEAN);
+}
+
+void tasks_run100Hz(void)
+{
+    WatchdogHandle *watchdog = hw_watchdog_allocateWatchdog();
+    hw_watchdog_initWatchdog(watchdog, RTOS_TASK_100HZ, 10U);
+
+    /* Infinite loop */
+    for (;;)
+    {
+        // test can
+        CanMsg msg = {
+            .std_id = 100,
+            .dlc    = 0,
+        };
+        io_can_pushTxMsgToQueue(&msg);
+
+        hw_watchdog_checkIn(watchdog);
+    }
+}
+
+void tasks_runCanTx(void)
+{
+    for (;;)
+    {
+        io_can_transmitMsgFromQueue();
+    }
+}
+
+void tasks_runCanRx(void)
+{
+    for (;;)
+    {
+        CanMsg rx_msg;
+        io_can_popRxMsgFromQueue(&rx_msg);
+    }
+}
```

Comment:
<img width="650" alt="image" src="https://github.com/UBCFormulaElectric/Consolidated-Firmware/assets/28274296/37d4ec10-3e54-460f-8019-42ae0d381986">

Messages get pushed to the queue when they're received, and then they should be handled in this thread after they're popped off the queue. However, after you pop the RX message off the queue, you're not currently actually doing anything with it. Add the JSONCAN lines from the image to update the RX table, and then the usual RX getters will work as expected.

Also needs to be done on VC.

Reply:
Unintentionally fixed on fsm!